tests/components/yale_smart_alarm/test_config_flow.py

```diff
@@ -0,0 +1,135 @@
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.yale_smart_alarm.config_flow.YaleSmartAlarmClient",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.yale_smart_alarm.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "username": "test-username",
+                "password": "test-password",
+                "name": "Yale Smart Alarm",
+                "area_id": "1",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "test-username"
+    assert result2["data"] == {
+        "username": "test-username",
+        "password": "test-password",
+        "name": "Yale Smart Alarm",
+        "area_id": "1",
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_invalid_auth(hass: HomeAssistant) -> None:
+    """Test we handle invalid auth."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.yale_smart_alarm.config_flow.YaleSmartAlarmClient",
+        side_effect=AuthenticationError,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "username": "test-username",
+                "password": "test-password",
+                "name": "Yale Smart Alarm",
+                "area_id": "1",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "invalid_auth"}
+
+
+async def test_import_flow_success(hass):
+    """Test a successful import of yaml."""
+    with patch(
+        "homeassistant.components.yale_smart_alarm.config_flow.YaleSmartAlarmClient",
+        return_value=True,
+    ), patch(
+        "homeassistant.components.yale_smart_alarm.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result2 = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={
+                "username": "test-username",
+                "password": "test-password",
+                "name": "Yale Smart Alarm",
+                "area_id": "1",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == "create_entry"
+    assert result2["title"] == "test-username"
+    assert result2["data"] == {
+        "username": "test-username",
+        "password": "test-password",
+        "name": "Yale Smart Alarm",
+        "area_id": "1",
+    }
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_import_flow_success_missing_requirement(hass):
```

Comment:
What's the purpose of this test?

Reply:
It was to check that the import to config_flow would be adding the eventually missing things from config.yaml during import. I got less test points until I added this into test-file

Reply:
What's the difference between this test and the test above?

Reply:
The difference is that this one "missing_requirements" check that the config flow automatically adds the possibly missing name and area_id key's into the config if they were missing in the config.yaml during import.

Reply:
Ok. Optionally these two tests could be combined by using pytest parametrization since only the import data and entry data changes:
https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions

Reply:
Have changed to using the parametrization